services/web-app/components/storybook-demo/storybook-demo.js

```diff
@@ -0,0 +1,144 @@
+import clsx from 'clsx'
+import PropTypes from 'prop-types'
+import { useState } from 'react'
+
+import * as styles from './storybook-demo.module.scss'
+
+const StorybookDemo = ({ tall, themeSelector, wide, url, variants }) => {
+  const columnWidth = wide ? 12 : 8
+
+  const demoClassNames = clsx(styles['storybook-demo'], {
+    [styles.tall]: tall,
+    [styles.wide]: wide
+  })
+
+  const themeItems = [
+    {
+      id: 'white',
+      label: 'White',
+      src: 'white'
+    },
+    {
+      id: 'g10',
+      label: 'Gray 10',
+      src: 'g10'
+    },
+    {
+      id: 'g90',
+      label: 'Gray 90',
+      src: 'g90'
+    },
+    {
+      id: 'g100',
+      label: 'Gray 100',
+      src: 'g100'
+    }
+  ]
+
+  const [theme, setTheme] = useState('')
+  const onThemeChange = (item) => {
+    setTheme(item.selectedItem.src)
+  }
+
+  const variantsDefined = typeof variants !== 'undefined' && variants.length > 1
+
+  // TODO: need to update this initialStateVarint --> need a default variant
+  const initialSetVariant = variantsDefined ? variants[1].variant : 'components-button--default'
+
+  const [variant, setVariant] = useState(initialSetVariant)
+
+  const iframeUrl = url + '/iframe.html?id=' + variant + '&globals=theme:' + theme
+
+  const onVariantChange = (item) => {
+    setVariant(item.selectedItem.variant)
+  }
+
+  // Only add border when theme and variant selectors are being displayed
+  const border = clsx({
+    [styles['theme-selector']]: variantsDefined
+  })
+
+  return (
+    <>
+      <Grid condensed>
+        {themeSelector && (
+          <Column sm={2} md={4}>
+            <Dropdown
+              titleText="Theme selector"
+              label="theme"
+              items={themeItems}
+              onChange={onThemeChange}
+              initialSelectedItem={themeItems[0]}
+              className={border}
+            />
+          </Column>
+        )}
+        {variantsDefined && (
+          <Column sm={2} md={4}>
+            <Dropdown
+              titleText="Variant selector"
+              label="variant"
+              items={variants}
+              initialSelectedItem={variants[1].label}
+              onChange={onVariantChange}
+            />
+          </Column>
+        )}
+      </Grid>
+      <Grid condensed>
+        <Column sm={4} md={8} lg={columnWidth} className={demoClassNames}>
+          <iframe
+            title="Component demo"
+            className={styles.iframe}
+            src={iframeUrl}
+            frameBorder="no"
+            allowtransparency="true"
+            allowFullScreen="true"
+          />
+        </Column>
+      </Grid>
+      <Grid>
+        <Column sm={4} md={7}>
+          <p className={styles.disclaimer}>
+            This live demo contains only a preview of functionality and styles available for this component.
+            View the <Link className={styles['disclaimer-link']} href="https://next.carbondesignsystem.com/">full demo</Link> on Storybook
```

Comment:
This link should be similar to the embed url but Replace `iframe.html?id=` with `?path=/story/`

Which makes me think we don't support this, and require the variant to be added to the variants prop. 
```

<StorybookDemo
  themeSelector={true}
  size="small"
  height="tall"
  url="https://react.carbondesignsystem.com/iframe.html?args=&id=components-button--default"
/>

```

Reply:
@francinelucca @alisonjoseph

Reply:
yep agreed!